### ISSUE TITLE
Why sample packing force finetuning to be on a single GPU

### ISSUE THREAD

**Issue by hahmad2008** [axolotl-ai-cloud/axolotl#1348]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/OpenAccess-AI-Collective/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

When I tried this commit from axolotl, I observed this behaviour.

For a single machine, If I have multiple GPUs and set `num_processes: n ` in the accelerate config that we need to use these GPUs. I have these two scenarios:

* If sample packing is True, then fine tuning is forced to be used/ see only single GPU! 
* If the sample packing is false, then fine tuning is running on all assigned GPUs, as expected.

### Current behaviour

When I tried this commit from axolotl, I observed this behaviour.

For a single machine, If I have multiple GPUs and set `num_processes: n ` in the accelerate config that we need to use these GPUs. I have these two scenarios:

* If sample packing is True, then fine tuning is forced to be used/ see only single GPU! 
* If the sample packing is false, then fine tuning is running on all assigned GPUs, as expected.

### Steps to reproduce

as above

### Config yaml

```yaml
### Config yaml


* finetune-config.yaml 


base_model: openlm-research/open_llama_3b_v2
base_model_config: openlm-research/open_llama_3b_v2
model_type: LlamaForCausalLM
tokenizer_type: LlamaTokenizer
trust_remote_code: null
is_falcon_derived_model: null
is_llama_derived_model: null
is_mistral_derived_model: null
load_in_8bit: false
load_in_4bit: true
strict: false
push_dataset_to_hub: null
datasets:
- path: mhenrichsen/alpaca_2k_test
  type: alpaca
dataset_prepared_path: /directory/fine_tune/test22_1/prepared-dataset
val_set_size: 0.01
adapter: qlora
lora_model_dir: null
sequence_len: 512
sample_packing: true
lora_r: 8
lora_alpha: 16
lora_dropout: 0.05
lora_target_modules: null
lora_target_linear: true
lora_fan_in_fan_out: null
wandb_project: null
wandb_entity: null
wandb_watch: null
wandb_run_id: null
wandb_log_model: null
output_dir: /directory/fine_tune/test22_1/model-finetuned
gradient_accumulation_steps: 1
micro_batch_size: 2
num_epochs: 1
optimizer: adamw_bnb_8bit
torchdistx_path: null
lr_scheduler: cosine
learning_rate: 0.0002
train_on_inputs: false
group_by_length: false
bf16: false
fp16: true
tf32: false
gradient_checkpointing: true
early_stopping_patience: null
resume_from_checkpoint: null
local_rank: null
logging_steps: 1
xformers_attention: null
flash_attention: null
gptq_groupsize: null
gptq_model_v1: null
warmup_steps: 10
eval_steps: 100
save_steps: 100
debug: null
deepspeed: null
weight_decay: 0.0
fsdp: null
fsdp_config: null
special_tokens:
  bos_token: <s>
  eos_token: </s>
  unk_token: <unk>
tokens: null



* accelerate-config.yaml 

```
compute_environment: LOCAL_MACHINE
distributed_type: 'NO'
downcast_bf16: 'no'
gpu_ids: all
machine_rank: 0
main_training_function: main
mixed_precision: fp16
num_machines: 1
num_processes: 2
rdzv_backend: static
same_network: true
tpu_env: []
tpu_use_cluster: false
tpu_use_sudo: false
use_cpu: false
```
```
```


### Possible solution

_No response_

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10

### axolotl branch-commit

c67fb7158312e47e3326f077f74485cf0a23b51a

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by NanoCode012** [comment]
Hello, I would recommend you delete the accelerate config. Let axolotl handle that. Try just set deepspeed config and let it run :)

**Comment by winglian** [comment]
unless you're running a multi-node setup, we don't recommend setting the accelerate config at all. Regardless though, multi-gpu should work with sample packing. 

I ran `accelerate launch -m axolotl.cli.train examples/mistral/qlora.yml  --eval_sample_packing=False` 

and I see 
```
# nvidia-smi 
Fri Mar  1 13:45:26 2024       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.129.03             Driver Version: 535.129.03   CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA RTX A6000               On  | 00000000:4F:00.0 Off |                  Off |
| 30%   62C    P2             296W / 300W |  18606MiB / 49140MiB |    100%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   1  NVIDIA RTX A6000               On  | 00000000:52:00.0 Off |                  Off |
| 36%   67C    P2             296W / 300W |  18590MiB / 49140MiB |    100%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
+---------------------------------------------------------------------------------------+
```
<img width="575" alt="Screenshot 2024-03-01 at 8 45 20 AM" src="https://github.com/OpenAccess-AI-Collective/axolotl/assets/381258/d01d9873-d736-4881-bdb4-df182445b638">
